client/components/forms/form-text-input/index.jsx

```diff
@@ -19,25 +18,31 @@ module.exports = React.createClass( {
 		};
 	},
 
-	render: function() {
-		var otherProps = omit( this.props, [ 'className', 'type' ] ),
-			classes = classNames( {
-				'form-text-input': true,
-				'is-error': this.props.isError,
-				'is-valid': this.props.isValid
-			} );
+	focus() {
+		this.refs.textField.focus();
+	},
+
+	render() {
+		const otherProps = omit( this.props, [ 'className', 'type', 'ref' ] );
```

Comment:
I think you could have left `ref` out, since I don't think `ref` is passed as a prop even when assigned in a parent component, and because you override it after applying the spread properties to the `<input >` element. In fact, this whole omit statement could be left out altogether -

`<input { ...this.props } ref="textField" type={ type } className={ classes } />`

It's a small detail, so not to hold up merge.